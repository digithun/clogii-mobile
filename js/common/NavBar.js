'use strict';

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
  Dimensions
} from 'react-native';

export const HEIGHT = 60;

export default class NavBar extends React.Component {
  render() {
    return (
      <View style={[styles.container, this.props.containerStyle]}>
        <View style={[styles.leftMenu, this.props.leftMenuStyle]}>
          {
            this.props.renderLeftMenu ? this.props.renderLeftMenu() : null
          }
        </View>
        <View style={[styles.titleMenu, this.props.titleStyle]}>
        {
          this.props.renderTitle ? this.props.renderTitle() : null
        }
        </View>
        <View style={[styles.rightMenu, this.props.rightMenuStyle]}>
          {
            this.props.renderRightMenu ? this.props.renderRightMenu() : null
          }
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 5,
    height: HEIGHT,
    width: Dimensions.get('window').width,
    backgroundColor: 'white',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center'
  },
  titleMenu: {
    alignItems: 'center'
  },
  leftMenu: {
    paddingLeft: 10,
    alignItems: 'flex-start'
  },
  rightMenu: {
    paddingRight: 10,
    alignItems: 'flex-end'
  }
});
