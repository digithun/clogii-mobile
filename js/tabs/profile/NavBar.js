'use strict';

import React from 'react';
import {
  Text,
  Platform,
  View,
  StyleSheet,
  TouchableOpacity,
  Image
} from 'react-native';

import {colors} from './common';

const NAV_BAR_HEIGHT = 50;

export default class NavBar extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.leftMenu} onPress={() => this.props.onLeftPress && this.props.onLeftPress()}>
          <Image style={styles.backButton} source={require('./img/icons/backButton.png')}/>
        </TouchableOpacity>
        <View style={styles.midMenu}>
          <Text style={styles.navText}>{this.props.title}</Text>
        </View>
        <TouchableOpacity style={styles.rightMenu} onPress={() => this.props.onRightPress && this.props.onRightPress()}>
          {
            this.props.renderRightMenu ? this.props.renderRightMenu() : null
          }
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 0,
    height: NAV_BAR_HEIGHT,
    backgroundColor: 'white',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderColor: colors.greyBorder
  },
  backButton: {
    height: 20,
    resizeMode: 'contain'
  },
  midMenu: {
    alignItems: 'center'
  },
  leftMenu: {
    paddingLeft: 20,
    alignItems: 'flex-start'
  },
  rightMenu: {
    paddingRight: 20,
    alignItems: 'flex-end'
  },
  navText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: 'rgb(209, 87, 92)'
  }
});
