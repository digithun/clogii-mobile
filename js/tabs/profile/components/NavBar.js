

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
} from 'react-native';

import { NavBarWithPinkButton } from '../../../common/NavBar';

import { colors } from '../../../common/styles';

export default class NavBar extends React.Component {
  render() {
    return (
      <NavBarWithPinkButton
        {...this.props}
        containerStyle={styles.container}
        renderRightMenu={this.renderRightMenu.bind(this)}
        title={this.props.title}
        titleTextStyle={styles.titleText}
      />
    );
  }

  renderRightMenu() {
    return (
      <TouchableOpacity style={styles.rightMenu} onPress={this.props.onRightPress}>
        {
          this.props.renderRightMenu ? this.props.renderRightMenu() : null
        }
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderColor: colors.greyBorder,
  },
  backButton: {
    height: 20,
    resizeMode: 'contain',
  },
  titleText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: colors.textPink,
  },
});
