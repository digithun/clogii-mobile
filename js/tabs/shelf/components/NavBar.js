'use strict';

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image
} from 'react-native';

import DefaultNavBar from '../../../common/NavBar';
export {HEIGHT} from '../../../common/NavBar';
import {colors} from '../../../common/styles';

export default class NavBar extends React.Component {
  render() {
    return (
      <DefaultNavBar
        backButton
        {...this.props}
        />
    );
  }
}
