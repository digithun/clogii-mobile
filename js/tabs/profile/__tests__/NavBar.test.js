import React from 'react';
import {
  Text,
  View,
  TouchableOpacity
} from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NavBar from '../NavBar';

describe('<NavBar/>', () => {
  it('render backButton by default', () => {
    const tree = renderer.create(<NavBar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render title', () => {
    const wrapper = shallow(<NavBar title="test title"/>);
    expect(wrapper.contains('test title')).toBe(true);
  });

  it('can custom rightMenu', () => {
    const wrapper = shallow(<NavBar renderRightMenu={() => <Text>RButton</Text>}/>)
    expect(wrapper.contains(<Text>RButton</Text>)).toBe(true);
  });

  it('handle onPress', () => {
    const leftFn = jest.fn();
    const rightFn = jest.fn();
    const wrapper = shallow(<NavBar renderRightMenu={() => <Text>RButton</Text>} onLeftPress={leftFn} onRightPress={rightFn}/>);
    const touchables = wrapper.find(TouchableOpacity);
    expect(touchables.length).toBeGreaterThan(0);
    touchables.at(0).simulate('press');
    expect(leftFn).toBeCalled();
    expect(rightFn).not.toBeCalled();
    touchables.at(1).simulate('press');
    expect(rightFn).toBeCalled();
  });
});

const A = props => (<TouchableOpacity onPress={props.onPress}></TouchableOpacity>)
