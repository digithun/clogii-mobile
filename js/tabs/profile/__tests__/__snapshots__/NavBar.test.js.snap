exports[`<NavBar/> render backButton by default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingTop": 5,
        "width": 750,
      },
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#eeeeee",
      },
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "paddingLeft": 10,
        },
        undefined,
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        } />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          },
          Object {
            "color": "rgb(209, 87, 92)",
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      } />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "paddingRight": 10,
        },
        undefined,
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined} />
  </View>
</View>
`;
